waku/waku_core/time.nim

```diff
@@ -10,7 +10,11 @@ import
 type Timestamp* = int64 # A nanosecond precision timestamp
 
 proc getNanosecondTime*[T: SomeNumber](timeInSeconds: T): Timestamp =
-  var ns = Timestamp(timeInSeconds * 1_000_000_000.T)
+  var ns: Timestamp
```

Comment:
seems more simple to change this function to `getNanosecondTime*(timeInSeconds: int64)`?

Reply:
Sure, I can do that. However, I was hesitant to make changes in a lot of places to the codebase because there are several instances in the code where different numeric types are passed to the function that will trigger `type mismatch` errors. That's why I preferred to keep it simple and just handle this problem within the function without casting types all over the project.